docs/guide/en/parameters-resolvers.md

```diff
@@ -0,0 +1,170 @@
+use Yiisoft\Input\Http\Attribute\Parameter\Query;
+
+final class PostController
+{
+    public function get(#[Query('id')] int $id): ResponseInterface
+    {
+        // ...
+        
+        /**
+         * @var int $id 
+         * @var ResponseInterface $response 
+         */
+        return $response;
+    }
+}
+```
+
+### Adding to middleware dispatcher
+
+An example of applying parameters' resolver to middleware dispatcher:
+
+```php
+use Psr\Container\ContainerInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Yiisoft\Hydrator\AttributeHandling\ParameterAttributesHandler;
+use Yiisoft\Hydrator\AttributeHandling\ResolverFactory\ContainerAttributeResolverFactory;
+use Yiisoft\Hydrator\HydratorInterface;
+use Yiisoft\Hydrator\TypeCaster\TypeCasterInterface;
+use Yiisoft\Input\Http\HydratorAttributeParametersResolver;
+use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
+use Yiisoft\Middleware\Dispatcher\MiddlewareFactory;
+use Yiisoft\Middleware\Dispatcher\ParametersResolverInterface;
+
+/**
+ * @var ContainerInterface $container
+ * @var EventDispatcherInterface $eventDispatcher
+ * @var TypeCasterInterface $typeCaster 
+ * @var HydratorInterface $hydrator
+ */
+$resolver = new HydratorAttributeParametersResolver(
+    new ParameterAttributesHandler(new ContainerAttributeResolverFactory($container)),
+    $typeCaster, // Optional, for customizing type casting
+    $hydrator, // Optional, for customizing hydration
+);
+$dispatcher = new MiddlewareDispatcher(new MiddlewareFactory($container, $resolver), $eventDispatcher);
+```
+
+More info about dependencies from [Yii Hydrator](https://github.com/yiisoft/hydrator) package can be found in its docs 
+and guide:
+ 
+- [attribute resolver factory](https://github.com/yiisoft/hydrator/blob/master/docs/guide/en/attribute-resolver-factory.md)
+- [type casting](https://github.com/yiisoft/hydrator/blob/master/docs/guide/en/typecasting.md) 
+- [hydrator](https://github.com/yiisoft/hydrator)
+
+Using within a group of resolvers:
+
+```php
+use Psr\Container\ContainerInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Yiisoft\Input\Http\HydratorAttributeParametersResolver;
+use Yiisoft\Middleware\Dispatcher\CompositeParametersResolver;
+use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
+use Yiisoft\Middleware\Dispatcher\MiddlewareFactory;
+use Yiisoft\Middleware\Dispatcher\ParametersResolverInterface;
+
+/** 
+ * @var ContainerInterface $container
+ * @var ParametersResolverInterface $existingResolver1
+ * @var ParametersResolverInterface $existingResolver2
+ * @var HydratorAttributeParametersResolver $addedResolver
+ * @var EventDispatcherInterface $eventDispatcher 
+ */
+$dispatcher = new MiddlewareDispatcher(
+    new MiddlewareFactory(
+        $container, 
+        new CompositeParametersResolver($existingResolver1, $existingResolver1, $addedResolver),
+    ),
+    $eventDispatcher,
+);
+```
+
+## `RequestInputParametersResolver`
```

Comment:
Add note about usage validating hydartor for request input DTO automatic validation.

Reply:
:heavy_check_mark: